rest-api/src/main/java/uk/gov/hmcts/ccd/definition/store/rest/endpoint/AccessTypeRolesController.java

```diff
@@ -0,0 +1,129 @@
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.PostMapping;
+import uk.gov.hmcts.ccd.definition.store.domain.service.EntityToResponseDTOMapper;
+import uk.gov.hmcts.ccd.definition.store.repository.AccessTypeRolesRepository;
+import uk.gov.hmcts.ccd.definition.store.repository.entity.AccessTypeRolesEntity;
+import uk.gov.hmcts.ccd.definition.store.repository.entity.CaseTypeEntity;
+import uk.gov.hmcts.ccd.definition.store.repository.entity.JurisdictionEntity;
+import uk.gov.hmcts.ccd.definition.store.repository.model.AccessTypeRolesField;
+import uk.gov.hmcts.ccd.definition.store.repository.model.OrganisationProfileIds;
+import uk.gov.hmcts.ccd.definition.store.repository.model.AccessTypeRolesJurisdictionResult;
+import uk.gov.hmcts.ccd.definition.store.repository.model.AccessTypeRolesJurisdictionResults;
+import uk.gov.hmcts.ccd.definition.store.repository.model.AccessTypeRolesResult;
+import uk.gov.hmcts.ccd.definition.store.repository.model.AccessTypeRolesRoleResult;
+
+import javax.validation.Valid;
+import javax.validation.constraints.NotNull;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@RestController
+@Api(value = "/api")
+@RequestMapping(value = "/api")
+public class AccessTypeRolesController {
+
+    private final EntityToResponseDTOMapper entityToResponseDTOMapper;
+    private final AccessTypeRolesRepository accessTypeRolesRepository;
+
+    @Autowired
+    public AccessTypeRolesController(EntityToResponseDTOMapper entityToResponseDTOMapper,
+                                     AccessTypeRolesRepository accessTypeRolesRepository) {
+        this.entityToResponseDTOMapper = entityToResponseDTOMapper;
+        this.accessTypeRolesRepository = accessTypeRolesRepository;
+    }
+
+
+    @PostMapping(value = "/retrieve-access-types",consumes = {"application/json"}, produces = {"application/json"})
+    @ApiOperation(value = "Get all accessTypes for all caseTypes", response = AccessTypeRolesJurisdictionResult.class,
+        responseContainer = "List")
+    @ApiResponses(value = {
+        @ApiResponse(code = 200, message = "List of valid access types request"),
+        @ApiResponse(code = 401, message = "Unauthorised request"),
+        @ApiResponse(code = 403, message = "Bad request")
+    })
+    public AccessTypeRolesJurisdictionResults retrieveAccessTypeRoles(
+        @Valid @RequestBody() @NotNull  OrganisationProfileIds organisationProfileIds) {
+        List<AccessTypeRolesField> accessTypeRoles = null;
+
+        List<AccessTypeRolesEntity>  accessTypeRolesEntities = accessTypeRolesRepository
+            .findByOrganisationProfileIds(organisationProfileIds.getOrganisationProfileIds());
+        accessTypeRoles =  accessTypeRolesEntities.stream().map(entityToResponseDTOMapper::map)
+            .collect(Collectors.toList());
+
```

Comment:
you also need to add case where organisationProfileIds is not sent - for this scenario all access type defs need to be returned. required for PRM batch process

Reply:
<img width="1075" alt="image" src="https://github.com/hmcts/ccd-definition-store-api/assets/137802184/63036b11-d31a-4306-9162-b736dcbdae8f">

if the body has nothing specified then the request will be a Bad Request (401).

Reply:
do we have a unit test that is testing this?